### ISSUE TITLE
LAPACK error on simple SVD

### ISSUE THREAD

**Issue by maximsch2** [JuliaLang/julia#15784]
```julia
X = ones(Float32, (19764, 13620));
svd(X)
```

throws an error:
```
 ** On entry to SGESDDPM parameter number 12 had an illegal value
ERROR: ArgumentError: invalid argument #12 to LAPACK call
 in gesdd! at linalg/lapack.jl:1482
 in svdfact! at linalg/svd.jl:17
 in svdfact at linalg/svd.jl:23
```

This sounds very close to https://github.com/scipy/scipy/issues/5401, maybe we need a similar workaround?

This is on julia 0.4.5 built from source on Ubuntu 14.04.

**Comment by andreasnoack** [comment]
Ha. That's a funny one. It should be easy to work around, though. Thanks for the SciPy people.

**Comment by maximsch2** [comment]
Changing places like https://github.com/JuliaLang/julia/blob/master/base/linalg/lapack.jl#L302 to 
```julia
lwork = BlasInt(nextfloat(real(work[1])))
```
should be enough, right?

**Comment by andreasnoack** [comment]
Almost. I think this would give an inexact error for smaller values so something like
```jl
lwork = round(BlasInt, nextfloat(real(work[1])))
```
should work. It would be great if you could make a PR.

**Comment by jianghaizhu** [comment]
Just tested it with `svd(rand(Float32, (9537, 9537)))`,  I still have this error.

>  ** On entry to SGESDDPM parameter number 12 had an illegal value
> ERROR: ArgumentError: invalid argument #12 to LAPACK call
>  in gesdd! at linalg/lapack.jl:1505
>  in svdfact! at linalg/svd.jl:17
>  in svdfact at linalg/svd.jl:23


 My julia version is

> Julia Version 0.4.6
> Commit 2e358ce (2016-06-19 17:16 UTC)
> Platform Info:
>   System: Darwin (x86_64-apple-darwin13.4.0)
>   CPU: Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
>   WORD_SIZE: 64
>   BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
>   LAPACK: libopenblas64_
>   LIBM: libopenlibm
>   LLVM: libLLVM-3.3

**Comment by andreasnoack** [comment]
It is fixed on master but it hasn't been backported yet. I forgot to label the fix for backporting but have done that now so it should be included in next bugfix release of 0.4.

**Comment by jianghaizhu** [comment]
Great! Thanks.
